torch/_inductor/cudagraph_trees.py

```diff
@@ -416,25 +422,41 @@ def cudagraphify(
     constants: Tuple[torch.Tensor, ...] = (),
     placeholders: Tuple[PlaceholderInfo, ...] = (),
     mutated_input_idxs: Tuple[int, ...] = (),
+    compile_id: Optional[CompileId] = None,
 ) -> Tuple[ModelType, OutputType]:
-    manager = get_container(device_index).get_tree_manager()
-    assert not (is_backward and is_inference)
-    mode = (
-        CompilationMode.BACKWARD
-        if is_backward
-        else (CompilationMode.INFERENCE if is_inference else CompilationMode.FORWARD)
-    )
+    # Even though this is technically runtime, we want to capture these overheads for
+    # "compilation metrics" logging.
+    metrics_context = get_metrics_context()
```

Comment:
The introduction of this metrics_context is what gives us the dynamo_compile scuba log that encompasses the "cudagraphify" and autotuning timings.

Reply:
<img width="695" alt="Screenshot 2024-12-03 at 5 24 59 PM" src="https://github.com/user-attachments/assets/99626b92-31b7-485c-bf59-f411b3e975b0">

Well what I'm seeing experimentally is something like this. i.e., I see the "CUDAGraphTreeManager.run_eager" timer that you added, but no "CUDAGraphTreeManager.record_function". And neither seem to account for the full time of the cudagraphify